docs/tools/prism-eslint-hook.js

```diff
@@ -0,0 +1,348 @@
+"use strict";
+
+const Prism = require("prismjs");
+const { docsExampleCodeToParsableCode } = require("./code-block-utils");
+
+let isAvailable = false;
+let Linter = null;
+let astUtils = null;
+
+try {
```

Comment:
Let's add a comment here explaining why we're testing.
```suggestion
/*
 * We can only do the syntax highlighting in the English-language
 * site because we need the Linter and astUtils. This same
 * code is run in translation sites where these utilities are
 * not available, so check to see if they are before attempting
 * to highlight the code.
 */
try {
```